modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2SaaSAppTokenRevocationTestCase.java

```diff
@@ -0,0 +1,152 @@
+                tenantAdminUsername, tenantAdminPassword);
+    }
+
+    @Test(alwaysRun = true, description = "Test validation of tokens issued to same tenant users " +
+            "when service provider is SaaS-disabled", groups = "wso2.is")
+    public void testSameTenantTokenValidationWhenSaasDisabled() throws Exception {
+
+        ServiceProvider serviceProvider = applicationManagementServiceClient.getApplication(SERVICE_PROVIDER_NAME);
+        serviceProvider.setSaasApp(false);
+        applicationManagementServiceClient.updateApplicationData(serviceProvider);
+        //at application update, token validation is done in async mode which may take some time at the server,
+        // hence adding a thread sleep....
+        Thread.sleep(500);
+        OAuth2TokenValidationRequestDTO requestDTO = new OAuth2TokenValidationRequestDTO();
+        OAuth2TokenValidationRequestDTO_OAuth2AccessToken accessTokenDTO = new
+                OAuth2TokenValidationRequestDTO_OAuth2AccessToken();
+        accessTokenDTO.setIdentifier(superTenantAccessToken);
+        accessTokenDTO.setTokenType(TOKEN_TYPE);
+        requestDTO.setAccessToken(accessTokenDTO);
+        OAuth2TokenValidationResponseDTO responseDTO = oauth2TokenValidationClient.validateToken(requestDTO);
+        assertEquals(responseDTO.getValid(), true,
+                "Token validation should be successful for the users of same tenant when " +
+                        "service provider is SaaS-disabled");
+    }
+
+    @Test(alwaysRun = true, description = "Test the validation of tokens issued to other tenants when " +
+            "service provider is SaaS-disabled", groups = "wso2.is")
+    public void testOtherTenantTokenValidationWhenSaasDisabled() throws Exception {
+
+        ServiceProvider serviceProvider = applicationManagementServiceClient.getApplication(SERVICE_PROVIDER_NAME);
+        serviceProvider.setSaasApp(false);
+        applicationManagementServiceClient.updateApplicationData(serviceProvider);
+        //at application update, token validation is done in async mode which may take some time at the server,
+        // hence adding a thread sleep....
+        Thread.sleep(500);
+        OAuth2TokenValidationRequestDTO requestDTO = new OAuth2TokenValidationRequestDTO();
+        OAuth2TokenValidationRequestDTO_OAuth2AccessToken accessTokenDTO = new
+                OAuth2TokenValidationRequestDTO_OAuth2AccessToken();
+        accessTokenDTO.setIdentifier(tenantAccessToken);
+        accessTokenDTO.setTokenType(TOKEN_TYPE);
+        requestDTO.setAccessToken(accessTokenDTO);
+        OAuth2TokenValidationResponseDTO responseDTO = oauth2TokenValidationClient.validateToken(requestDTO);
+        assertEquals(responseDTO.getValid(), false,
+                "Token validation should fail for users of other tenants when " +
+                        "service provider is SaaS-disabled");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testClear() throws Exception {
```

Comment:
This is not a *Test* method. 
Any Before/After Method/Class should not be start with a prefix *test*. 
Prefix *test* is to signify that we are testing something.